frontend/src/app/utils/rsvHelper.ts

```diff
@@ -0,0 +1,45 @@
+type SupportedDisease = {
+  supportedDisease: {
+    name: string;
+    __typename: "SupportedDisease";
+  };
+};
+export type DeviceWithoutModelOrManufacturer = Omit<
+  FacilityFormDeviceType,
+  "model" | "manufacturer"
+>;
+
+function filterRsvFromSingleDevice(
+  device: DeviceWithoutModelOrManufacturer | FacilityFormDeviceType
+) {
+  const supportedDiseaseArray =
+    device.supportedDiseaseTestPerformed as SupportedDisease[];
+
+  // no supportedDiseaseTestPerformed defined due to any casting, return empty array
+  if (supportedDiseaseArray === undefined) return [];
+
+  const supportedDiseases = supportedDiseaseArray.map(
+    (sd) => sd.supportedDisease.name
+  );
+  if (supportedDiseases.includes("RSV")) {
+    return supportedDiseaseArray.filter(
+      (d) => d.supportedDisease.name !== "RSV"
+    );
+  }
+  return supportedDiseaseArray;
+}
+
+type DeviceType = DeviceWithoutModelOrManufacturer | FacilityFormDeviceType;
+
+export function filterRsvFromAllDevices<AmbiguousDeviceType extends DeviceType>(
```

Comment:
for everyone's situational awareness: I extended this filtering function with a generic that accepts / filters down both the full `FacilityFormDeviceType` and the subset device type in the test card that doesn't have the model or manufacterer info. This way, whichever is passed in is also passed out, allowing us to remove the `@ts-ignore` override that was there in [the previous commit](https://github.com/CDCgov/prime-simplereport/pull/6545/commits/8de673cd5b8214d83c1ef32dd8bd9e7b3ca0c561#diff-02aa88a48980d7ec2ec531bd5c4a68a527613d878cad9470fbcf5d5d903dc523R757)

Reply:
~~**suggestion:** could we change this type to use `QueriedDeviceType` exported by `QueueItem.tsx` so we don't have to maintain additional types?~~

edit: hmm nevermind on this, just realized the issue with ManageDevices using a different type than QueueItem so it may be more trouble than it's worth to resolve that